plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/DiskCache.kt

```diff
@@ -98,12 +103,21 @@ class DiskCache(
 
     override fun invalidateClientRegistration(ssoRegion: String) {
         LOG.debug { "invalidateClientRegistration for $ssoRegion" }
+        InMemoryCache.remove(clientRegistrationCache(ssoRegion).toString())
         clientRegistrationCache(ssoRegion).tryDeleteIfExists()
     }
 
     override fun loadClientRegistration(cacheKey: ClientRegistrationCacheKey): ClientRegistration? {
         LOG.debug { "loadClientRegistration for $cacheKey" }
-        val inputStream = clientRegistrationCache(cacheKey).tryInputStreamIfExists()
+        val cacheFile = clientRegistrationCache(cacheKey)
+        // try InMemoryCacheFirst in case of stale registration on full disk
```

Comment:
shouldn't the disk be the source of truth?
consider the case where multiple different JetBrains IDE instances are running